torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -334,5 +334,27 @@ Tensor CrossEntropyLossImpl::forward(
     options.reduction());
 }
 
+// ============================================================================
+
+BCEWithLogitsLossImpl::BCEWithLogitsLossImpl(
+  const BCEWithLogitsLossOptions& options_) : options(options_) {
+  reset();
+}
+
+void BCEWithLogitsLossImpl::reset() {
+  register_buffer("weight", options.weight());
+  register_buffer("pos_weight", options.pos_weight());
```

Comment:
In Python, I think registering a buffer makes it available through class member accessor such as `module.weight` and `module.pos_weight`, and to mimic this in C++ version I think we might want to have `weight` and `pos_weight` in the `BCEWithLogitsLossImpl` class, and here we can do
```cpp
weight = register_buffer("weight", options.weight());
pos_weight = register_buffer("pos_weight", options.pos_weight());
```